src/main/java/org/broadinstitute/hellbender/tools/exome/CalculateCoverageComponents.java

```diff
@@ -0,0 +1,135 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.ArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ExomeAnalysisProgramGroup;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SparkToggleCommandLineProgram;
+import org.broadinstitute.hellbender.utils.hdf5.HDF5File;
+import org.broadinstitute.hellbender.utils.pca.PCA;
+import org.broadinstitute.hellbender.utils.svd.SVDFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Perform PCA on coverage counts.
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        summary = "Perform PCA on coverage counts",
+        oneLineSummary = "Perform PCA on coverage counts",
+        programGroup = ExomeAnalysisProgramGroup.class)
+public class CalculateCoverageComponents extends SparkToggleCommandLineProgram {
```

Comment:
gradle test gives warning: [serial] serializable class CalculateCoverageComponents has no definition of serialVersionUID